### ISSUE TITLE
Accessibility: ToolStrip: TabStop= false, pressing Tab or Shift-Tab doesn’t work after clicked toolStripButton/SplitButton/DropDownButton

### ISSUE THREAD

**Issue by Olina-Zhang** [dotnet/winforms#5794]
* .NET Core Version: 
.Net SDK 6.0.100-rc.2.21466.17

* Have you experienced this same bug with .NET Framework?: 
Yes

**Minimal repro:**
1. Create a Winforms .net Core application with ToolStrip added, keeping its TabStop false by default
2. Add some toolstrip items: ToolStripButton, ToolStripDropDownButton, ToolStripSplitButton, ToolStripTextBox, ToolStripComboBox
3. Build and run application
4. Mouse click any toolStripItems:  ToolStripButton/ToolStripDropDownButton/ToolStripSplitButton, then press keyboard: Tab or Shift-Tab
[ToolStripSample.zip](https://github.com/dotnet/winforms/files/7183156/ToolStripSample.zip)

**Actual behavior:**
No response, cannot go through toolStrip items inside ToolStrip in order using Tab or reverse order using Shift-tab, but if clicking ToolStripTextBox/ToolStripComboBox, then Tab or Shift-Tab works well
![TabIssue1](https://user-images.githubusercontent.com/26474449/133735290-bfb2d290-146b-46e0-a147-e5474c47afd9.gif)
**Expected behavior:**
Can go through toolStrip items inside ToolStrip in order using Tab or reverse order using Shift-tab whenever focusing on which one

**Comment by ArtemTatarinov** [comment]
This is happening because not `ToolStrip` nor `ToolStripButton` has focus after click was performed on the `ToolStripButton`. If `ToolStripButton` was clicked, then `ToolStrip` is not gaining focus (if it wasn't focused) or is losing it (if it was focused). And `ToolStripButton` itself can't be focused (it doesn't have `Focus` method, unlike `ToolStripTextBox` and other `ToolStripControlHost` successors).

In this example the `Form` has the focus and the `ToolStrip` has not before and after the click event is being processed:
![image](https://user-images.githubusercontent.com/87859299/133984777-08822162-85d6-4043-80de-b2ad7f4bf973.png)
![image](https://user-images.githubusercontent.com/87859299/133984480-786cc809-df4a-46a5-a93b-569e9df19388.png)

So when we press `Tab` after clicking `ToolStripButton` `ToolStrip` isn't reacting to it, because it isn't focused and doesn't handle keyboard events.

We can change this behavior by focusing `ToolStrip` at the button click event handlers.

**ToolStripButton**
```
protected override void OnClick(EventArgs e)
        {
            if (CheckOnClick)
            {
                Checked = !Checked;
            }

            base.OnClick(e);

            Parent.Focus(); // Add this row
        }
```

**ToolStripSplitButton**
```
 protected virtual void OnButtonClick(EventArgs e)
        {
            DefaultItem?.FireEvent(ToolStripItemEventType.Click);
            ((EventHandler)Events[s_eventButtonClick])?.Invoke(this, e);

            Parent.Focus(); // Add this row
        }
```

**ToolStripDropDownItem**
```
//  Add this method
protected override void OnClick(EventArgs e)
        {
            base.OnClick(e);

            Parent.Focus();
        }
```

But maybe it will change `ToolStrip` behavior too drastically? Maybe the current behavior is expected and shouldn't be changed? @merriemcgaw, could you please tell what do you think about it?

**Comment by merriemcgaw** [comment]
@Tanya-Solyanik your thoughts? I have to think about this a little. When you get to the `ToolStripDropDownButton` via keyboard the user would expect to hit the `space` perhaps to activate it, and then use `Down Arrow` to enter the `ToolStripDropDownMenu`. Then arrow keys to move up and down the list? What do you think?

**Comment by Tanya-Solyanik** [comment]
In this case ToolStrip was created with TabStop property set to false, This ensures that it does not get focus. TextBoxes have to have focus because otherwise user would not be able to type. Buttons and variations of buttons probably are not focusable for historical reasons, i.e. toolstip was designed for mouse access initially. 

Assuming "tabstop==true" case works as expected, this is "by design".

**Comment by merriemcgaw** [comment]
Thanks @Tanya-Solyanik I can go with that.

**Comment by merriemcgaw** [comment]
> Assuming "tabstop==true" case works as expected, this is "by design".

@ArtemTatarinov can you confirm and then close this?

**Comment by ArtemTatarinov** [comment]
Yes, clicking `ToolStrip` buttons with `TabStop`==`true` doesn't affect the current focus and moving focus with `Tab` key continued to work like no clicking was performed at all:

- The first `ToolStrip` was focused, clicked `ToolStripButton` at the first `ToolStrip`, then pressed `Tab` two times - the focus was moved to the second `ToolStrip` and then to the third one
![tabAfteClick1](https://user-images.githubusercontent.com/87859299/134513796-c44aa73c-655f-466d-975a-f17e9e5c67b0.gif)

- The second `ToolStrip` was focused, clicked button at the first `ToolStrip`, the second `ToolStrip` saved its focus, pressed `Tab` and the focus was moved from the second `ToolStrip` to the third one
![tabAfteClick2](https://user-images.githubusercontent.com/87859299/134516623-216f6d18-e49f-479f-8f4b-8f790cf54b03.gif)

- Set focus to the third `ToolStrip`, then moved it to the first `ToolStrip` using `Tab` key, clicked buttons at the second and the third `ToolStrips`, the first `ToolStrip still has its focus, pressed `Tab` and moved to the second `ToolStrip`
![tabAfteClick3](https://user-images.githubusercontent.com/87859299/134518740-577624b2-4133-497a-b163-96287eb17ed6.gif)
